securedrop/journalist_app/api2/types.py

```diff
@@ -0,0 +1,97 @@
+from dataclasses import dataclass, field
+from enum import IntEnum, StrEnum, auto
+from typing import (
+    Any,
+    Dict,
+    List,
+    NewType,
+    Set,
+    Union,
+)
+
+Version = NewType("Version", str)
+
+
+# TODO: generic UUID[T] in Python 3.12
+SourceUUID = NewType("SourceUUID", str)
+ItemUUID = NewType("ItemUUID", str)
+JournalistUUID = NewType("JournalistUUID", str)
+
+
+EventID = NewType("EventID", str)  # int, but opaque on the wire
+
+
+class EventType(StrEnum):
+    REPLY_SENT = auto()
+
+
+class EventStatusCode(IntEnum):
```

Comment:
Not 100% sold on using HTTP status codes...or at least would be nice if we could also send some sort of more specific error message if we have one (e.g. in EventHandler.process() we're losing the underlying exception reason)